src/rules/non-interactive-tooltip-trigger.js

```diff
@@ -0,0 +1,126 @@
+  const child = jsxNode.children
+  const childName = getJSXOpeningElementName(child.openingElement)
+
+  // First check specific requirements for anchor
+  if (childName === 'a' && !isInteractiveAnchor(child)) {
+    messageId = 'anchorTagWithoutHref'
+    return {messageId, node: jsxNode}
+  }
+  // Then check specific requirements input
+  if (childName === 'input' && !isInteractiveInput(child)) {
+    messageId = 'hiddenInput'
+    return {messageId, node: jsxNode}
+  }
+  // Then check if the child is interactive
+  if (!isInteractive(child)) {
+    // If child is not interactive, check if there are any grandchildren that is interactive
+    const hasJsxGrands =
+      child.children.length > 0 && child.children.filter(gChild => gChild.type === 'JSXElement').length > 0 //.some(gChild => isInteractive(gChild))
+
+    if (!hasJsxGrands) {
+      messageId = 'nonInteractiveTrigger'
+    } else {
+      const hasInteractiveGrands = child.children // is there any way I can access all child nodes? :/
+        .filter(gChild => gChild.type === 'JSXElement')
+        .some(gChild => {
+          const gChildName = getJSXOpeningElementName(gChild.openingElement)
+          // TODO: How can I check all child nodes?
+          return checkTriggerElement(gChild).messageId === ''
+        })
+      if (!hasInteractiveGrands) messageId = 'nonInteractiveTrigger'
+    }
+
+    return {messageId, node: jsxNode}
+  }
+
+  // All good the element is interactive
+  return {messageId, node: jsxNode}
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      url:
+        'https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/tree/HEAD/docs/rules/no-noninteractive-element-interactions.md',
+      description: 'Non-interactive elements should not be assigned mouse or keyboard event listeners.'
+    },
+    schema: [
+      {
+        properties: {
+          skipImportCheck: {
+            type: 'boolean'
+          }
+        }
+      }
+    ],
+    messages: {
+      nonInteractiveTrigger:
+        'The `Tooltip` component expects a single React element that contains interactive content. Consider using a `<button>` or equivalent interactive element instead.',
+      anchorTagWithoutHref:
+        'Anchor tags without an href attribute are not be considered interactive, therefore can not be used as a trigger for a tooltip.',
+      hiddenInput:
+        'Hidden inputs are not be considered interactive, therefore can not be used as a trigger for a tooltip.',
+      singleChild: 'The `Tooltip` component expects a single React element as a child.'
+    }
+  },
+  create(context) {
+    const stack = []
+    const {options} = context
+    return {
+      JSXElement(jsxNode) {
+        // If `skipImportCheck` is true, this rule will check for direct slot children
+        // in any components (not just ones that are imported from `@primer/react`).
+        const skipImportCheck = context.options[0] ? context.options[0].skipImportCheck : false
```

Comment:
I am unsure what this `skipImportCheck` does. Do I need it here or not necessary maybe in this rule? 👀

Reply:
@broccolinisoup one guess (I'm also not sure) is that we need a way to verify that the `<Tooltip>` component comes from the `@primer/react` package and is not a local `Tooltip` for the project. Skipping it may be useful in `primer/react` since we can assume that `Tooltip` is the correct `Tooltip` and the import check would fail since we don't import from `@primer/react`.

Reply:
Oh I see! That makes sense now. Thank you!